src/components/AssetMatching/components/ActionsToolbox/ActionsToolbox.tsx

```diff
@@ -19,8 +19,9 @@ const ActionsToolbox: FC<Props> = ({
   matchingActions = [],
   buttonStyles,
 }) => {
-  const primaryActions = matchingActions.slice(0, 2);
-  const secondaryActions = matchingActions.slice(2, matchingActions.length);
+  const primaryActions = matchingActions.length > 3 ? matchingActions.slice(0, 2) : matchingActions;
```

Comment:
This works fine! 

Optional: Alternatively, you can render primary actions directly to avoid extra checks (perf improvement). 

```
const shouldDisplayOnlyPrimaryActions = matchingActions?.length > 0 && matchingActions?.length <= 3
if (shouldDisplayOnlyPrimaryActions) {
    return (
      <ActionsToolboxWrapper customActionsContent={customActionsContent}>
        <PrimaryActions {...(buttonStyles || {})} primaryActions={matchingActions} />
      </ActionsToolboxWrapper>
    );
  }

 const primaryActions = matchingActions.slice(0, 2);
 const secondaryActions = matchingActions.slice(2, matchingActions.length);

return the same as before...

....
// you can also refactor this.
const ActionsToolboxWrapper: FC<{ customActionsContent?: JSX.Element | null }> = ({
  children,
  customActionsContent,
}) => {
  return (
    <div css={flex}>
      {children}
      {customActionsContent && customActionsContent}
    </div>
  );
};
```